pkg/jwt/github/client.go

```diff
@@ -0,0 +1,94 @@
+package github
+
+import (
+	"context"
+	"net/http"
+
+	"github.com/google/go-github/github"
+)
+
+// Client contains the methods that abstract an API
+type Client interface {
+	CurrentUser(*http.Client) (string, error)
+	Organizations(*http.Client) ([]string, error)
+	Teams(*http.Client) (OrganizationTeams, error)
+}
+
+type client struct {
+	baseURL string
+}
+
+// NewClient creates a new instance of client
+func NewClient() Client {
+	return &client{}
+}
+
+// OrganizationTeams is a map of organization names and teams
+type OrganizationTeams map[string][]string
+
+// CurrentUser retrieves the current authenticated user for an http client
+func (c *client) CurrentUser(httpClient *http.Client) (string, error) {
+	client := github.NewClient(httpClient)
+
+	currentUser, _, err := client.Users.Get(context.TODO(), "")
+	if err != nil {
+		return "", err
+	}
+
+	return *currentUser.Login, nil
+}
+
+// Teams retrieves the teams that the authenticated user belongs
+func (c *client) Teams(httpClient *http.Client) (OrganizationTeams, error) {
+	client := github.NewClient(httpClient)
```

Comment:
This could have being created on the constructor `NewClient(httpClient *http.Client)`

Reply:
No, it couldn't. The constructor is called on start up time, the Teams is called on each authentication requests with different `http` contexts, which means you have to create a new GitHub client for each request that will contain different contexts

Reply:
Got it 👍